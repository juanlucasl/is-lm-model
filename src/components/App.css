/* SPDX-License-Identifier: Apache-2.0 */
.App {
  margin: 0 auto;
  max-width: max-content;
}

.App__functions-section {
  display: flex;
  flex-direction: row;
  gap: 1rem;
}

.App__variables-section {
  display: flex;
  flex-direction: column;
  gap: 2rem;
}

.App__anchor-icons {
  bottom: 3rem;
  display: flex;
  float: right;
  gap: 0.5rem;
  position: relative;
  right: 2rem;
}

.App__anchor-icons svg {
  fill: black;
  font-size: 1.6rem;
  max-height: 1.6rem;
  max-width: 1.6rem;
  vertical-align: text-bottom;
}
