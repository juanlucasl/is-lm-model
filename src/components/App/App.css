/* SPDX-License-Identifier: Apache-2.0 */
.App {
  margin: 0 auto;
  max-width: max-content;
}

.App__functions-section {
  display: flex;
  flex-direction: row;
  gap: 1rem;
}

.App__variables-section {
  display: flex;
  flex-direction: column;
  gap: 2rem;
}

.App__anchors {
  bottom: 3rem;
  display: flex;
  float: right;
  gap: 0.5rem;
  position: relative;
  right: 2rem;
}

.App__anchor-icon {
  color: black;
  fill: black;
  font-size: 1.6rem;
  height: 1.6rem;
  vertical-align: text-bottom;
  width: 1.6rem;
}

.App__anchor-icon:active {
  transform: scale(0.8);
}
