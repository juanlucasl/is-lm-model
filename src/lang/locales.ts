/* SPDX-License-Identifier: Apache-2.0 */
export enum Locales {
  English = "en",
  Spanish = "es",
}
