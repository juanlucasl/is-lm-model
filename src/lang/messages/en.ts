const CONSUMPTION_UP = "&uarr; Consumption (C)";
const CONSUMPTION_DOWN = "&darr; Consumption (C)";

const INTEREST_RATE_UP = "&uarr; Interest Rate (i)";
const INTEREST_RATE_DOWN = "&darr; Interest Rate (i)";

const OUTPUT_UP = "&uarr; Output or GDP (Y) (In equilibrium Y = Z (Demand for goods))";
const OUTPUT_DOWN = "&darr; Output or GDP (Y) (In equilibrium Y = Z (Demand for goods))";

const IS_RIGHT = "&rarr; The IS Curve shifts to the right";
const IS_LEFT = "&larr; The IS Curve shifts to the left";

const LM_RIGHT = "&rarr; The LM Curve shifts to the right";
const LM_LEFT = "&larr; The LM Curve shifts to the left";

const en = {
  IS_LM_MODEL: "IS-LM Model",
  IS_CURVE: "IS Curve",
  LM_CURVE: "LM Curve",
  TAXES: "Taxes (T)",
  INVESTMENT: "Investment (I)",
  PUBLIC_EXPENDITURE: "Public Expenditure (G)",
  MONEY_SUPPLY: "Money Supply (M)",
  TAXES_INCREASE_DESCRIPTION: `
    <p>
      The government charges more taxes and people have less surplus income to demand goods
      <ul>
        <li>${CONSUMPTION_DOWN}</li>
        <li>${INTEREST_RATE_DOWN}</li>
        <li>${OUTPUT_DOWN}</li>
        <li>${IS_LEFT}</li>
      </ul>
    </p>
  `,
  TAXES_DECREASE_DESCRIPTION: `
    <p>
      The government charges less taxes and people have more surplus income to demand goods
      <ul>
        <li>${CONSUMPTION_UP}</li>
        <li>${INTEREST_RATE_UP}</li>
        <li>${OUTPUT_UP}</li>
        <li>${IS_RIGHT}</li>
      </ul>
    </p>
  `,
  INVESTMENT_INCREASE_DESCRIPTION: `
    <p>
      Higher demand for capital goods, hence higher Aggregate Demand
      <ul>
        <li>${CONSUMPTION_UP}</li>
        <li>${INTEREST_RATE_UP}</li>
        <li>${OUTPUT_UP}</li>
        <li>${IS_RIGHT}</li>
      </ul>
    </p>
  `,
  INVESTMENT_DECREASE_DESCRIPTION: `
    <p>
      Lower demand for capital goods, hence lower Aggregate Demand
      <ul>
        <li>${CONSUMPTION_DOWN}</li>
        <li>${INTEREST_RATE_DOWN}</li>
        <li>${OUTPUT_DOWN}</li>
        <li>${IS_LEFT}</li>
      </ul>
    </p>
  `,
  PUBLIC_EXPENDITURE_INCREASE_DESCRIPTION: `
    <p>
      The government demands more goods and services, hence higher Aggregate Demand
      <ul>
        <li>${CONSUMPTION_UP}</li>
        <li>${INTEREST_RATE_UP}</li>
        <li>${OUTPUT_UP}</li>
        <li>${IS_RIGHT}</li>
      </ul>
    </p>
  `,
  PUBLIC_EXPENDITURE_DECREASE_DESCRIPTION: `
    <p>
      The government demands fewer goods and services, hence lower Aggregate Demand
      <ul>
        <li>${CONSUMPTION_DOWN}</li>
        <li>${INTEREST_RATE_DOWN}</li>
        <li>${OUTPUT_DOWN}</li>
        <li>${IS_LEFT}</li>
      </ul>
    </p>
  `,
  MONEY_SUPPLY_INCREASE_DESCRIPTION: `
    <p>
      Increased money supply drives down the price of money, reflected in interest rates
      <ul>
        <li>${INTEREST_RATE_DOWN}</li>
        <li>${LM_RIGHT}</li>
      </ul>
    </p>
  `,
  MONEY_SUPPLY_DECREASE_DESCRIPTION: `
    <p>
      Lower money supply drives up the price of money, reflected in interest rates.
      <ul>
        <li>${INTEREST_RATE_UP}</li>
        <li>${LM_LEFT}</li>
      </ul>
    </p>
  `,
};

export default en;
