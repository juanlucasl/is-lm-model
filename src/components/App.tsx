/* SPDX-License-Identifier: Apache-2.0 */
import "./App.css";
import React, { useState } from "react";
import { FormattedMessage, useIntl } from "react-intl";
import { MarkGithubIcon } from "@primer/octicons-react";
import { Siapache } from "@icons-pack/react-simple-icons";
import FunctionsDrawer from "./FunctionsDrawer";
import Variables from "./Variables";
import {
  DEMANDA_BASE_DE_DINERO,
  CONSUMO_AUTONOMO,
  PROPENSION_MARGINAL_A_CONSUMIR,
  SENSIBILIDAD_DE_LA_DEMANDA_MONETARIA_AL_INTERES,
} from "../constants";
import { Variable } from "../models";

function App() {
  const intl = useIntl();
  /**
   *
   * Inversion (I): Sumatoria de inversión no residual (la compra de nuevas máquinas/plantas
   * por parte de las empresas) y de inversión residual (la compra de apartamentos/viviendas
   * por parte de los individuos).
   *
   * Gasto Publico (G): Bienes y servicios comprados por el Estado.
   *
   * Impuestos (I): Impuestos cobrados por el Estado.
   *
   * Oferta Monetaria (M): Es la suma de los billetes y monedas en poder del público, más los
   * depósitos a la vista en el sistema bancario.
   */

  const [inversion, setInversion] = useState(50);
  const [gastoPublico, setGastoPublico] = useState(50);
  const [impuestos, setImpuestos] = useState(50);

  const [ofertaMonetaria, setOfertaMonetaria] = useState(50);

  const funcionConsumo = (renta: number) =>
    CONSUMO_AUTONOMO + PROPENSION_MARGINAL_A_CONSUMIR * (renta - impuestos);

  const funcionIS = (produccion: number) =>
    funcionConsumo(-produccion) + inversion + gastoPublico;

  const funcionLM = (produccion: number) =>
    DEMANDA_BASE_DE_DINERO +
    produccion * SENSIBILIDAD_DE_LA_DEMANDA_MONETARIA_AL_INTERES -
    ofertaMonetaria;

  const variablesIS: Variable[] = [
    {
      value: impuestos,
      setValue: setImpuestos,
      label: intl.formatMessage({ id: "TAXES" }),
      increaseDescription: intl.formatMessage({ id: "TAXES_INCREASE_DESCRIPTION" }),
      decreaseDescription: intl.formatMessage({ id: "TAXES_DECREASE_DESCRIPTION" }),
    },
    {
      value: inversion,
      setValue: setInversion,
      label: intl.formatMessage({ id: "INVESTMENT" }),
      increaseDescription: intl.formatMessage({ id: "INVESTMENT_INCREASE_DESCRIPTION" }),
      decreaseDescription: intl.formatMessage({ id: "INVESTMENT_DECREASE_DESCRIPTION" }),
    },
    {
      value: gastoPublico,
      setValue: setGastoPublico,
      label: intl.formatMessage({ id: "PUBLIC_EXPENDITURE" }),
      increaseDescription: intl.formatMessage({
        id: "PUBLIC_EXPENDITURE_INCREASE_DESCRIPTION",
      }),
      decreaseDescription: intl.formatMessage({
        id: "PUBLIC_EXPENDITURE_DECREASE_DESCRIPTION",
      }),
    },
  ];

  const variablesLM: Variable[] = [
    {
      value: ofertaMonetaria,
      setValue: setOfertaMonetaria,
      label: intl.formatMessage({ id: "MONEY_SUPPLY" }),
      increaseDescription: intl.formatMessage({ id: "MONEY_SUPPLY_INCREASE_DESCRIPTION" }),
      decreaseDescription: intl.formatMessage({ id: "MONEY_SUPPLY_DECREASE_DESCRIPTION" }),
    },
  ];

  return (
    <article className="App">
      <h1>
        <FormattedMessage id="IS_LM_MODEL" />
      </h1>
      <section className="App__functions-section">
        <FunctionsDrawer
          functionsToDraw={[funcionIS, funcionLM]}
          verticalAxisLabel="i"
          horizontalAxisLabel="Y"
        />
        <section className="App__variables-section">
          <Variables
            variables={variablesIS}
            title={intl.formatMessage({ id: "IS_CURVE" })}
            description="Y = C(Y-T) + I(Y, i) + G"
          />
          <Variables
            variables={variablesLM}
            title={intl.formatMessage({ id: "LM_CURVE" })}
            description="M/P = Y ⋅ L(i)"
          />
        </section>
      </section>
      <div className="App__anchor-icons">
        <a href="https://github.com/juanlucasl/is-lm-model" target="_blank" rel="noreferrer">
          <MarkGithubIcon aria-label="Source code in Github"/>
        </a>
        <a href="https://www.apache.org/licenses/LICENSE-2.0" target="_blank" rel="noreferrer">
          <Siapache aria-label="Apache License 2.0" title="Apache License 2.0"/>
        </a>
      </div>
    </article>
  );
}

export default App;
