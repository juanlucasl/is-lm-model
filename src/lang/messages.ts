import en from "./messages/en";
import es from "./messages/es";

export const MESSAGES = {
  en,
  es,
}
