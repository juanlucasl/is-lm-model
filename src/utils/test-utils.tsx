/* SPDX-License-Identifier: Apache-2.0 */
import { ReactElement } from "react";
import { IntlProvider } from "react-intl";
import { render, RenderResult } from "@testing-library/react";
import { Locales } from "../lang/locales";
import { MESSAGES } from "../lang/messages";

export function renderWithIntlProvider(ui: ReactElement, locale: Locales): RenderResult {
  return render(
    <IntlProvider locale={locale} messages={MESSAGES[locale]}>
      {ui}
    </IntlProvider>
  );
}
