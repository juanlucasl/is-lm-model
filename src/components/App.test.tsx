/* SPDX-License-Identifier: Apache-2.0 */
import React from "react";
import { screen } from "@testing-library/react";
import App from "./App";
import { MESSAGES } from "../lang/messages";
import { Locales } from "../lang/locales";
import { renderWithIntlProvider } from "../utils/test-utils";

describe("App", () => {
  it("renders correctly with English locale", () => {
    const { container } = renderWithIntlProvider(<App />, Locales.English);
    expect(screen.getByText(MESSAGES[Locales.English].IS_LM_MODEL)).toBeVisible();
    expect(container).toMatchSnapshot();
  });

  it("renders correctly with Spanish locale", () => {
    const { container } = renderWithIntlProvider(<App />, Locales.Spanish);
    expect(screen.getByText(MESSAGES[Locales.Spanish].IS_LM_MODEL)).toBeVisible();
    expect(container).toMatchSnapshot();
  });
});
