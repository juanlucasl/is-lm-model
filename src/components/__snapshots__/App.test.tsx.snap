// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<div>
  <article
    class="App"
  >
    <h1>
      IS-LM Model
    </h1>
    <section
      class="App__functions-section"
    >
      <canvas
        aria-label="IS-LM"
        class="functions-drawer"
        height="500"
        role="img"
        width="500"
      />
      <section
        class="App__variables-section"
      >
        <section
          class="variables"
        >
          <div
            class="variables__header"
          >
            <h2>
              IS Curve
            </h2>
            <var>
              Y = C(Y-T) + I(Y, i) + G
            </var>
          </div>
          <div
            class="variable-input"
          >
            <span>
              Taxes (T)
            </span>
            <div
              aria-disabled="false"
              aria-label="+ Taxes (T)"
              class="arrow arrow__upwards "
              data-testid="arrow-increase"
              data-tooltip-html="
    <p>
      The government charges more taxes and people have less surplus income to demand goods
      <ul>
        <li>&darr; Consumption (C)</li>
        <li>&darr; Interest Rate (i)</li>
        <li>&darr; Output or GDP (Y) (In equilibrium Y = Z (Demand for goods))</li>
        <li>&larr; The IS Curve shifts to the left</li>
      </ul>
    </p>
  "
              id="taxest-increase"
              role="button"
            >
              ↑
            </div>
            <div
              aria-disabled="false"
              aria-label="- Taxes (T)"
              class="arrow arrow__downwards "
              data-testid="arrow-decrease"
              data-tooltip-html="
    <p>
      The government charges less taxes and people have more surplus income to demand goods
      <ul>
        <li>&uarr; Consumption (C)</li>
        <li>&uarr; Interest Rate (i)</li>
        <li>&uarr; Output or GDP (Y) (In equilibrium Y = Z (Demand for goods))</li>
        <li>&rarr; The IS Curve shifts to the right</li>
      </ul>
    </p>
  "
              id="taxest-decrease"
              role="button"
            >
              ↓
            </div>
          </div>
          <div
            class="variable-input"
          >
            <span>
              Investment (I)
            </span>
            <div
              aria-disabled="false"
              aria-label="+ Investment (I)"
              class="arrow arrow__upwards "
              data-testid="arrow-increase"
              data-tooltip-html="
    <p>
      Higher demand for capital goods, hence higher Aggregate Demand
      <ul>
        <li>&uarr; Consumption (C)</li>
        <li>&uarr; Interest Rate (i)</li>
        <li>&uarr; Output or GDP (Y) (In equilibrium Y = Z (Demand for goods))</li>
        <li>&rarr; The IS Curve shifts to the right</li>
      </ul>
    </p>
  "
              id="investmenti-increase"
              role="button"
            >
              ↑
            </div>
            <div
              aria-disabled="false"
              aria-label="- Investment (I)"
              class="arrow arrow__downwards "
              data-testid="arrow-decrease"
              data-tooltip-html="
    <p>
      Lower demand for capital goods, hence lower Aggregate Demand
      <ul>
        <li>&darr; Consumption (C)</li>
        <li>&darr; Interest Rate (i)</li>
        <li>&darr; Output or GDP (Y) (In equilibrium Y = Z (Demand for goods))</li>
        <li>&larr; The IS Curve shifts to the left</li>
      </ul>
    </p>
  "
              id="investmenti-decrease"
              role="button"
            >
              ↓
            </div>
          </div>
          <div
            class="variable-input"
          >
            <span>
              Public Expenditure (G)
            </span>
            <div
              aria-disabled="false"
              aria-label="+ Public Expenditure (G)"
              class="arrow arrow__upwards "
              data-testid="arrow-increase"
              data-tooltip-html="
    <p>
      The government demands more goods and services, hence higher Aggregate Demand
      <ul>
        <li>&uarr; Consumption (C)</li>
        <li>&uarr; Interest Rate (i)</li>
        <li>&uarr; Output or GDP (Y) (In equilibrium Y = Z (Demand for goods))</li>
        <li>&rarr; The IS Curve shifts to the right</li>
      </ul>
    </p>
  "
              id="publicexpenditureg-increase"
              role="button"
            >
              ↑
            </div>
            <div
              aria-disabled="false"
              aria-label="- Public Expenditure (G)"
              class="arrow arrow__downwards "
              data-testid="arrow-decrease"
              data-tooltip-html="
    <p>
      The government demands fewer goods and services, hence lower Aggregate Demand
      <ul>
        <li>&darr; Consumption (C)</li>
        <li>&darr; Interest Rate (i)</li>
        <li>&darr; Output or GDP (Y) (In equilibrium Y = Z (Demand for goods))</li>
        <li>&larr; The IS Curve shifts to the left</li>
      </ul>
    </p>
  "
              id="publicexpenditureg-decrease"
              role="button"
            >
              ↓
            </div>
          </div>
        </section>
        <section
          class="variables"
        >
          <div
            class="variables__header"
          >
            <h2>
              LM Curve
            </h2>
            <var>
              M/P = Y ⋅ L(i)
            </var>
          </div>
          <div
            class="variable-input"
          >
            <span>
              Money Supply (M)
            </span>
            <div
              aria-disabled="false"
              aria-label="+ Money Supply (M)"
              class="arrow arrow__upwards "
              data-testid="arrow-increase"
              data-tooltip-html="
    <p>
      Increased money supply drives down the price of money, reflected in interest rates
      <ul>
        <li>&darr; Interest Rate (i)</li>
        <li>&rarr; The LM Curve shifts to the right</li>
      </ul>
    </p>
  "
              id="moneysupplym-increase"
              role="button"
            >
              ↑
            </div>
            <div
              aria-disabled="false"
              aria-label="- Money Supply (M)"
              class="arrow arrow__downwards "
              data-testid="arrow-decrease"
              data-tooltip-html="
    <p>
      Lower money supply drives up the price of money, reflected in interest rates.
      <ul>
        <li>&uarr; Interest Rate (i)</li>
        <li>&larr; The LM Curve shifts to the left</li>
      </ul>
    </p>
  "
              id="moneysupplym-decrease"
              role="button"
            >
              ↓
            </div>
          </div>
        </section>
      </section>
    </section>
  </article>
</div>
`;

exports[`App renders correctly with English locale 1`] = `
<div>
  <article
    class="App"
  >
    <h1>
      IS-LM Model
    </h1>
    <section
      class="App__functions-section"
    >
      <canvas
        aria-label="IS-LM"
        class="functions-drawer"
        height="500"
        role="img"
        width="500"
      />
      <section
        class="App__variables-section"
      >
        <section
          class="variables"
        >
          <div
            class="variables__header"
          >
            <h2>
              IS Curve
            </h2>
            <var>
              Y = C(Y-T) + I(Y, i) + G
            </var>
          </div>
          <div
            class="variable-input"
          >
            <span>
              Taxes (T)
            </span>
            <div
              aria-disabled="false"
              aria-label="+ Taxes (T)"
              class="arrow arrow__upwards "
              data-testid="arrow-increase"
              data-tooltip-html="
    <p>
      The government charges more taxes and people have less surplus income to demand goods
      <ul>
        <li>&darr; Consumption (C)</li>
        <li>&darr; Interest Rate (i)</li>
        <li>&darr; Output or GDP (Y) (In equilibrium Y = Z (Demand for goods))</li>
        <li>&larr; The IS Curve shifts to the left</li>
      </ul>
    </p>
  "
              id="taxest-increase"
              role="button"
            >
              ↑
            </div>
            <div
              aria-disabled="false"
              aria-label="- Taxes (T)"
              class="arrow arrow__downwards "
              data-testid="arrow-decrease"
              data-tooltip-html="
    <p>
      The government charges less taxes and people have more surplus income to demand goods
      <ul>
        <li>&uarr; Consumption (C)</li>
        <li>&uarr; Interest Rate (i)</li>
        <li>&uarr; Output or GDP (Y) (In equilibrium Y = Z (Demand for goods))</li>
        <li>&rarr; The IS Curve shifts to the right</li>
      </ul>
    </p>
  "
              id="taxest-decrease"
              role="button"
            >
              ↓
            </div>
          </div>
          <div
            class="variable-input"
          >
            <span>
              Investment (I)
            </span>
            <div
              aria-disabled="false"
              aria-label="+ Investment (I)"
              class="arrow arrow__upwards "
              data-testid="arrow-increase"
              data-tooltip-html="
    <p>
      Higher demand for capital goods, hence higher Aggregate Demand
      <ul>
        <li>&uarr; Consumption (C)</li>
        <li>&uarr; Interest Rate (i)</li>
        <li>&uarr; Output or GDP (Y) (In equilibrium Y = Z (Demand for goods))</li>
        <li>&rarr; The IS Curve shifts to the right</li>
      </ul>
    </p>
  "
              id="investmenti-increase"
              role="button"
            >
              ↑
            </div>
            <div
              aria-disabled="false"
              aria-label="- Investment (I)"
              class="arrow arrow__downwards "
              data-testid="arrow-decrease"
              data-tooltip-html="
    <p>
      Lower demand for capital goods, hence lower Aggregate Demand
      <ul>
        <li>&darr; Consumption (C)</li>
        <li>&darr; Interest Rate (i)</li>
        <li>&darr; Output or GDP (Y) (In equilibrium Y = Z (Demand for goods))</li>
        <li>&larr; The IS Curve shifts to the left</li>
      </ul>
    </p>
  "
              id="investmenti-decrease"
              role="button"
            >
              ↓
            </div>
          </div>
          <div
            class="variable-input"
          >
            <span>
              Public Expenditure (G)
            </span>
            <div
              aria-disabled="false"
              aria-label="+ Public Expenditure (G)"
              class="arrow arrow__upwards "
              data-testid="arrow-increase"
              data-tooltip-html="
    <p>
      The government demands more goods and services, hence higher Aggregate Demand
      <ul>
        <li>&uarr; Consumption (C)</li>
        <li>&uarr; Interest Rate (i)</li>
        <li>&uarr; Output or GDP (Y) (In equilibrium Y = Z (Demand for goods))</li>
        <li>&rarr; The IS Curve shifts to the right</li>
      </ul>
    </p>
  "
              id="publicexpenditureg-increase"
              role="button"
            >
              ↑
            </div>
            <div
              aria-disabled="false"
              aria-label="- Public Expenditure (G)"
              class="arrow arrow__downwards "
              data-testid="arrow-decrease"
              data-tooltip-html="
    <p>
      The government demands fewer goods and services, hence lower Aggregate Demand
      <ul>
        <li>&darr; Consumption (C)</li>
        <li>&darr; Interest Rate (i)</li>
        <li>&darr; Output or GDP (Y) (In equilibrium Y = Z (Demand for goods))</li>
        <li>&larr; The IS Curve shifts to the left</li>
      </ul>
    </p>
  "
              id="publicexpenditureg-decrease"
              role="button"
            >
              ↓
            </div>
          </div>
        </section>
        <section
          class="variables"
        >
          <div
            class="variables__header"
          >
            <h2>
              LM Curve
            </h2>
            <var>
              M/P = Y ⋅ L(i)
            </var>
          </div>
          <div
            class="variable-input"
          >
            <span>
              Money Supply (M)
            </span>
            <div
              aria-disabled="false"
              aria-label="+ Money Supply (M)"
              class="arrow arrow__upwards "
              data-testid="arrow-increase"
              data-tooltip-html="
    <p>
      Increased money supply drives down the price of money, reflected in interest rates
      <ul>
        <li>&darr; Interest Rate (i)</li>
        <li>&rarr; The LM Curve shifts to the right</li>
      </ul>
    </p>
  "
              id="moneysupplym-increase"
              role="button"
            >
              ↑
            </div>
            <div
              aria-disabled="false"
              aria-label="- Money Supply (M)"
              class="arrow arrow__downwards "
              data-testid="arrow-decrease"
              data-tooltip-html="
    <p>
      Lower money supply drives up the price of money, reflected in interest rates.
      <ul>
        <li>&uarr; Interest Rate (i)</li>
        <li>&larr; The LM Curve shifts to the left</li>
      </ul>
    </p>
  "
              id="moneysupplym-decrease"
              role="button"
            >
              ↓
            </div>
          </div>
        </section>
      </section>
    </section>
  </article>
</div>
`;

exports[`App renders correctly with Spanish locale 1`] = `
<div>
  <article
    class="App"
  >
    <h1>
      Modelo IS-LM
    </h1>
    <section
      class="App__functions-section"
    >
      <canvas
        aria-label="IS-LM"
        class="functions-drawer"
        height="500"
        role="img"
        width="500"
      />
      <section
        class="App__variables-section"
      >
        <section
          class="variables"
        >
          <div
            class="variables__header"
          >
            <h2>
              Relación IS
            </h2>
            <var>
              Y = C(Y-T) + I(Y, i) + G
            </var>
          </div>
          <div
            class="variable-input"
          >
            <span>
              Impuestos (T)
            </span>
            <div
              aria-disabled="false"
              aria-label="+ Impuestos (T)"
              class="arrow arrow__upwards "
              data-testid="arrow-increase"
              data-tooltip-html="
    <p>
      El Estado cobra más impuestos y la gente tiene menor excedente de renta para demandar
      bienes
      <ul>
        <li>&darr; Consumo (C)</li>
        <li>&darr; Tipo de Interés (i)</li>
        <li>&darr; Producción o PIB (Y) (En equilibrio Y = Z (Demanda de bienes))</li>
        <li>&larr; La Curva IS se desplaza hacia la izquierda</li>
      </ul>
    </p>
  "
              id="impuestost-increase"
              role="button"
            >
              ↑
            </div>
            <div
              aria-disabled="false"
              aria-label="- Impuestos (T)"
              class="arrow arrow__downwards "
              data-testid="arrow-decrease"
              data-tooltip-html="
    <p>
      El Estado cobra menos impuestos y la gente tiene mayor excedente de renta para demandar
      bienes
      <ul>
        <li>&uarr; Consumo (C)</li>
        <li>&uarr; Tipo de Interés (i)</li>
        <li>&uarr; Producción o PIB (Y) (En equilibrio Y = Z (Demanda de bienes))</li>
        <li>&rarr; La Curva IS se desplaza hacia la derecha</li>
      </ul>
    </p>
  "
              id="impuestost-decrease"
              role="button"
            >
              ↓
            </div>
          </div>
          <div
            class="variable-input"
          >
            <span>
              Inversión (I)
            </span>
            <div
              aria-disabled="false"
              aria-label="+ Inversión (I)"
              class="arrow arrow__upwards "
              data-testid="arrow-increase"
              data-tooltip-html="
    <p>
      Mayor demanda de bienes de capital, por lo tanto, es mayor la Demanda Agregada
      <ul>
        <li>&uarr; Consumo (C)</li>
        <li>&uarr; Tipo de Interés (i)</li>
        <li>&uarr; Producción o PIB (Y) (En equilibrio Y = Z (Demanda de bienes))</li>
        <li>&rarr; La Curva IS se desplaza hacia la derecha</li>
      </ul>
    </p>
  "
              id="inversini-increase"
              role="button"
            >
              ↑
            </div>
            <div
              aria-disabled="false"
              aria-label="- Inversión (I)"
              class="arrow arrow__downwards "
              data-testid="arrow-decrease"
              data-tooltip-html="
    <p>
      Menor demanda de bienes de capital, por lo tanto, es menor la Demanda Agregada
      <ul>
        <li>&darr; Consumo (C)</li>
        <li>&darr; Tipo de Interés (i)</li>
        <li>&darr; Producción o PIB (Y) (En equilibrio Y = Z (Demanda de bienes))</li>
        <li>&larr; La Curva IS se desplaza hacia la izquierda</li>
      </ul>
    </p>
  "
              id="inversini-decrease"
              role="button"
            >
              ↓
            </div>
          </div>
          <div
            class="variable-input"
          >
            <span>
              Gasto Público (G)
            </span>
            <div
              aria-disabled="false"
              aria-label="+ Gasto Público (G)"
              class="arrow arrow__upwards "
              data-testid="arrow-increase"
              data-tooltip-html="
    <p>
      El Estado demanda más bienes y servicios, por lo tanto, es mayor la Demanda Agregada
      <ul>
        <li>&uarr; Consumo (C)</li>
        <li>&uarr; Tipo de Interés (i)</li>
        <li>&uarr; Producción o PIB (Y) (En equilibrio Y = Z (Demanda de bienes))</li>
        <li>&rarr; La Curva IS se desplaza hacia la derecha</li>
      </ul>
    </p>
  "
              id="gastopblicog-increase"
              role="button"
            >
              ↑
            </div>
            <div
              aria-disabled="false"
              aria-label="- Gasto Público (G)"
              class="arrow arrow__downwards "
              data-testid="arrow-decrease"
              data-tooltip-html="
    <p>
      El Estado demanda menos bienes y servicios, por lo tanto, es menor la Demanda Agregada
      <ul>
        <li>&darr; Consumo (C)</li>
        <li>&darr; Tipo de Interés (i)</li>
        <li>&darr; Producción o PIB (Y) (En equilibrio Y = Z (Demanda de bienes))</li>
        <li>&larr; La Curva IS se desplaza hacia la izquierda</li>
      </ul>
    </p>
  "
              id="gastopblicog-decrease"
              role="button"
            >
              ↓
            </div>
          </div>
        </section>
        <section
          class="variables"
        >
          <div
            class="variables__header"
          >
            <h2>
              Relación LM
            </h2>
            <var>
              M/P = Y ⋅ L(i)
            </var>
          </div>
          <div
            class="variable-input"
          >
            <span>
              Oferta Monetaria (M)
            </span>
            <div
              aria-disabled="false"
              aria-label="+ Oferta Monetaria (M)"
              class="arrow arrow__upwards "
              data-testid="arrow-increase"
              data-tooltip-html="
    <p>
      Mayor oferta de dinero hace bajar su precio, reflejado en el tipo de interés
      <ul>
        <li>&darr; Tipo de Interés (i)</li>
        <li>&rarr; La Curva LM se desplaza hacia la derecha</li>
      </ul>
    </p>
  "
              id="ofertamonetariam-increase"
              role="button"
            >
              ↑
            </div>
            <div
              aria-disabled="false"
              aria-label="- Oferta Monetaria (M)"
              class="arrow arrow__downwards "
              data-testid="arrow-decrease"
              data-tooltip-html="
    <p>
      Menor oferta de dinero hace subir su precio, reflejado en el tipo de interés
      <ul>
        <li>&uarr; Tipo de Interés (i)</li>
        <li>&larr; La Curva LM se desplaza hacia la izquierda</li>
      </ul>
    </p>
  "
              id="ofertamonetariam-decrease"
              role="button"
            >
              ↓
            </div>
          </div>
        </section>
      </section>
    </section>
  </article>
</div>
`;
